src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -688,82 +689,72 @@ bool getDesuperHtrInput(EnergyPlusData &state)
                                           ErrorsFound,
                                           state.dataIPShortCut->cCurrentModuleObject + " Name");
 
-        state.dataWaterThermalTanks->WaterHeaterDesuperheater(DesuperheaterNum).Name = state.dataIPShortCut->cAlphaArgs(1);
-        state.dataWaterThermalTanks->WaterHeaterDesuperheater(DesuperheaterNum).Type = state.dataIPShortCut->cCurrentModuleObject;
+        auto &DesupHtr = state.dataWaterThermalTanks->WaterHeaterDesuperheater(DesuperheaterNum);
+
+        DesupHtr.Name = state.dataIPShortCut->cAlphaArgs(1);
+        DesupHtr.Type = state.dataIPShortCut->cCurrentModuleObject;
```

Comment:
New local reference to `state.dataWaterThermalTanks->WaterHeaterDesuperheater(DesuperheaterNum)` throughout for readability.